api/src/main/java/com/capitalone/dashboard/service/DeployServiceImpl.java

```diff
@@ -257,4 +272,110 @@ private EnvironmentComponent createEnvComponent(CollectorItem collectorItem, Dep
 
         return environmentComponentRepository.save(deploy); // Save = Update (if ID present) or Insert (if ID not there)
     }
+
+    @Override
+    public String createRundeckBuild(Document doc, Map<String, String[]> parameters, String executionId, String status) throws HygieiaException {
+        Node executionNode = doc.getElementsByTagName("execution").item(0);
+        Node jobNode = executionNode.getFirstChild();
+        RundeckXMLParser p = new RundeckXMLParser(doc);
+        DeployDataCreateRequest request = new DeployDataCreateRequest();
+        request.setExecutionId(executionId);
+        request.setDeployStatus(status.toUpperCase());
+        String appNameOption = evaluateParametersOrDefault(parameters, p, "appName", false, "appName", "hygieiaAppName"); 
+        if (appNameOption == null) {
+            appNameOption = getAttributeValue(executionNode, "project");
+        }
+        request.setAppName(appNameOption);
+        request.setEnvName(evaluateParametersOrDefault(parameters, p, "envName", true, "environment", "envName", 
+                "env", "hygieiaEnvName"));
+        request.setArtifactName(evaluateParametersOrDefault(parameters, p, "artifactName", true,"artifactId", "artifactName", "hygieiaArtifactName"));
+        request.setArtifactGroup(evaluateParametersOrDefault(parameters, p, "artifactGroup", false,"artifactGroup", "group", "hygieiaArtifactGroup"));
+        request.setArtifactVersion(evaluateParametersOrDefault(parameters, p, "artifactVersion", false,"version", "artifactVersion"));
+        request.setNiceName(evaluateParametersOrDefault(parameters, p, "niceName", false,"niceName", "hygieiaNiceName"));
```

Comment:
@jmcshane  - instead of limiting the values for default options, can we do a match on regex pattern for keywords "artifact", "group","version" for  ArtifactName, ArtifactGroup and ArtifactVersion attributes.